rust/processor/src/models/ans_models/ans_lookup.rs

```diff
@@ -0,0 +1,776 @@
+                                    e.registered_address = name_record.registered_address.clone();
+                                    // name_record.expiration_timestamp may be zero if primary name wsc occurs before name record wsc,
+                                    // so return the larger of the two timestamps
+                                    e.expiration_timestamp = cmp::max(
+                                        name_record.expiration_timestamp,
+                                        e.expiration_timestamp,
+                                    );
+                                    // name_record.is_primary may be false here, so use OR to preserve is_primary status
+                                    e.is_primary = name_record.is_primary || e.is_primary;
+                                    e.is_deleted = name_record.is_deleted;
+                                })
+                                .or_insert(name_record);
+                        }
+                    },
+                    Err(e) => {
+                        tracing::error!(
+                            "Failed to parse primary name reverse lookup table item. Error: {:?}",
+                            e
+                        );
+                        continue;
+                    },
+                }
+            }
+        }
+        (new_current_ans_lookups, new_ans_lookups)
+    }
+
+    // Parse the primary name reverse record from write table item.
+    // The table key is the target address the primary name points to.
+    // The table value data has the domain and subdomain of the primary name.
+    pub fn from_primary_name_record_table_write_table_item_v1(
+        write_table_item: &WriteTableItem,
+        txn_version: i64,
+        txn_timestamp: chrono::NaiveDateTime,
+        address_to_primary_name_record: &AddressToPrimaryNameRecord,
+        conn: &mut PgPoolConnection,
+    ) -> anyhow::Result<(
+        HashMap<CurrentAnsLookupPK, Self>,
+        HashMap<AnsLookupPK, AnsLookup>,
+    )> {
+        let mut new_current_ans_lookups = HashMap::new();
+        let mut new_ans_lookups = HashMap::new();
+
+        let table_item_data = write_table_item.data.as_ref().unwrap();
+        let key_type = table_item_data.key_type.as_str();
+        let key = serde_json::from_str(&table_item_data.key).unwrap_or_else(|e| {
+            panic!(
+                "version {} failed! failed to parse type {}, data {:?}. Error: {:?}",
+                txn_version, key_type, table_item_data.key, e
+            )
+        });
+        let value_type: &str = table_item_data.value_type.as_str();
+        let value = &table_item_data.value;
+        let primary_name_record: NameRecordKeyV1 =
+            serde_json::from_str(value).unwrap_or_else(|e| {
+                panic!(
+                    "version {} failed! failed to parse type {}, data {:?}. Error: {:?}",
+                    txn_version, value_type, value, e
+                )
```

Comment:
we have a better pattern for this: 
tracing::error!(transaction_version=txn_version..., "failed to parse NameRecordKeyV1")
panic("failed to parse NameRecordKeyV1")